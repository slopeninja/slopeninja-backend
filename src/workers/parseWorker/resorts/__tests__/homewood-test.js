import fs from 'fs';
import {
  parseHomewoodSnow,
  parseHomewoodLiftCounts,
  parseHomewoodLifts,
  parseHomewoodTrails,
  parseHomewoodTrailCounts,
} from '../homewood';
import { createHtmlParser } from '../../parserFactory';

test('fetches Homewood snow data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/homewood-weather.html.input`);

  const resortData = await createHtmlParser('snow', parseHomewoodSnow)(htmlText);
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: 12,
      snowDepthBase: 53,
      snowDepthSummit: 96,
    },
  });
});

test('fetches all null for nonexisting snow values', async () => {
  const resortData = await createHtmlParser('snow', parseHomewoodSnow)('<html></html>');
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: null,
      snowDepthBase: null,
      snowDepthSummit: null,
    },
  });
});

test('fetches Homewood lifts data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/homewood-weather.html.input`);
  const resortData = await createHtmlParser('liftCounts', parseHomewoodLiftCounts)(htmlText);
  expect(resortData).toEqual({
    liftCounts: {
      total: 8,
      open: 6,
    },
  });
});

test('fetches all null for nonexisting lift values', async () => {
  const resortData = await createHtmlParser('liftCounts', parseHomewoodLiftCounts)('<html></html>');
  expect(resortData).toEqual({
    liftCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Homewood trails data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/homewood-weather.html.input`);
  const resortData = await createHtmlParser('trailCounts', parseHomewoodTrailCounts)(htmlText);
  expect(resortData).toEqual({
    trailCounts: {
      total: 67,
      open: 63,
    },
  });
});

test('fetches all null for nonexisting trails values', async () => {
  const resortData = await createHtmlParser('trailCounts', parseHomewoodTrailCounts)('<html></html>');
  expect(resortData).toEqual({
    trailCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Homewood lift list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/homewood-weather.html.input`);
  const resortData = await createHtmlParser('lifts', parseHomewoodLifts)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('lifts', parseHomewoodLifts)('<html></html>');
  expect(resortData).toMatchObject({ lifts: [] });
});

test('fetches Homewood trail list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/homewood-weather.html.input`);
  const resortData = await createHtmlParser('trails', parseHomewoodTrails)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('trails', parseHomewoodTrails)('<html></html>');
  expect(resortData).toMatchObject({ trails: [] });
});
