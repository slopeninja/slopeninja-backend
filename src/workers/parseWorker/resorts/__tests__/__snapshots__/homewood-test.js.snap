// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches Homewood lift list correctly 1`] = `
Object {
  "lifts": Array [
    Object {
      "category": null,
      "name": "Madden Chair",
      "status": null,
    },
    Object {
      "category": null,
      "name": "Quail Chair",
      "status": null,
    },
    Object {
      "category": null,
      "name": "Old Homewood Express",
      "status": null,
    },
    Object {
      "category": null,
      "name": "South Carpet",
      "status": null,
    },
    Object {
      "category": null,
      "name": "Ellis Chair",
      "status": null,
    },
    Object {
      "category": null,
      "name": "Magic Carpet",
      "status": null,
    },
    Object {
      "category": null,
      "name": "Happy Platter",
      "status": null,
    },
    Object {
      "category": null,
      "name": "Alpine Platter",
      "status": null,
    },
  ],
}
`;

exports[`fetches Homewood trail list correctly 1`] = `
Object {
  "trails": Array [
    Object {
      "category": "Madden Chair",
      "level": "intermediate",
      "name": "Shaft",
      "status": "open",
    },
    Object {
      "category": "Madden Chair",
      "level": "intermediate",
      "name": "Tailings",
      "status": "open",
    },
    Object {
      "category": "Madden Chair",
      "level": "intermediate",
      "name": "Last Resort",
      "status": "open",
    },
    Object {
      "category": "Madden Chair",
      "level": "intermediate",
      "name": "Ore Car",
      "status": "open",
    },
    Object {
      "category": "Madden Chair",
      "level": "intermediate",
      "name": "Lombard Street",
      "status": "open",
    },
    Object {
      "category": "Madden Chair",
      "level": "intermediate",
      "name": "Northern Return",
      "status": "open",
    },
    Object {
      "category": "Madden Chair",
      "level": "intermediate",
      "name": "Chute",
      "status": "open",
    },
    Object {
      "category": "Madden Chair",
      "level": "intermediate",
      "name": "Cradle",
      "status": "open",
    },
    Object {
      "category": "Madden Chair",
      "level": "advanced",
      "name": "Jimmy's Run",
      "status": "open",
    },
    Object {
      "category": "Madden Chair",
      "level": "advanced",
      "name": "Race Course",
      "status": "open",
    },
    Object {
      "category": "Madden Chair",
      "level": "advanced",
      "name": "The Face",
      "status": "open",
    },
    Object {
      "category": "Madden Chair",
      "level": "advanced",
      "name": "Glory Hole",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "beginner",
      "name": "Homeward Bound",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "beginner",
      "name": "Overload",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "intermediate",
      "name": "Sunnyside",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "intermediate",
      "name": "Jug Handle",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "intermediate",
      "name": "Martin's Lane",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "intermediate",
      "name": "Short Cut",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "intermediate",
      "name": "Prospector",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "intermediate",
      "name": "El Capitan",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "intermediate",
      "name": "Mighty Fine",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "advanced",
      "name": "Exhibition",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "advanced",
      "name": "Double Trouble",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "advanced",
      "name": "Spillway",
      "status": "open",
    },
    Object {
      "category": "Quail Chair",
      "level": "advanced",
      "name": "Drain Pipe",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "intermediate",
      "name": "Lake Louise",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "intermediate",
      "name": "Bonanza",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "intermediate",
      "name": "Kolby's Escape",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "intermediate",
      "name": "Pot O' Gold",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "intermediate",
      "name": "Wash Pan",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "intermediate",
      "name": "Miner's Delight",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "advanced",
      "name": "Nugget",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "advanced",
      "name": "Lasky Lane",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "advanced",
      "name": "Juniper",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "advanced",
      "name": "Lower Lake Louise",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "advanced",
      "name": "Gilbert's Gulch",
      "status": "open",
    },
    Object {
      "category": "Old Homewood Express",
      "level": "advanced",
      "name": "Woody Fellers",
      "status": "open",
    },
    Object {
      "category": "South Carpet",
      "level": "beginner",
      "name": "Children's Center",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "beginner",
      "name": "Upper Homeward Bound",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "beginner",
      "name": "Rainbow Ridge",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "intermediate",
      "name": "Hidden Vein",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "intermediate",
      "name": "Second Creek",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "intermediate",
      "name": "Hoop's",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "intermediate",
      "name": "The Glades",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "intermediate",
      "name": "Lower Ego Alley",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "intermediate",
      "name": "Cassandra's Crossing",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "intermediate",
      "name": "Smooth Cruise",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "intermediate",
      "name": "Third Creek",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "intermediate",
      "name": "High Grade",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "intermediate",
      "name": "Noonchester Traverse",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "advanced",
      "name": "Cedar Ridge",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "advanced",
      "name": "White Lightning",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "advanced",
      "name": "Dutch Treat",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "advanced",
      "name": "Big Dipper",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "advanced",
      "name": "Show Off",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "advanced",
      "name": "Ivory Face",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "advanced",
      "name": "Hobbit Land",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "advanced",
      "name": "Blow Off",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "advanced",
      "name": "Little Dipper",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "advanced",
      "name": "The Shoulder",
      "status": "closed",
    },
    Object {
      "category": "Ellis Chair",
      "level": "advanced",
      "name": "Ego Alley",
      "status": "open",
    },
    Object {
      "category": "Ellis Chair",
      "level": "expert",
      "name": "55' Chutes",
      "status": "closed",
    },
    Object {
      "category": "Ellis Chair",
      "level": "expert",
      "name": "Wally's Folly",
      "status": "closed",
    },
    Object {
      "category": "Ellis Chair",
      "level": "expert",
      "name": "Main Cirque",
      "status": "closed",
    },
    Object {
      "category": "Magic Carpet",
      "level": "beginner",
      "name": "Easy Slider",
      "status": "open",
    },
    Object {
      "category": "Happy Platter",
      "level": "beginner",
      "name": "Joyride",
      "status": "open",
    },
    Object {
      "category": "Alpine Platter",
      "level": "beginner",
      "name": "Smooth Ride",
      "status": "open",
    },
  ],
}
`;
